### PR TITLE
PT-BR/Translation for what-is-json.md (@thierrysantoos1)

### PR DESCRIPTION
**Pull request by 0xthierry**
Hi everyone,

This PR translates to portuguese the page what-is-json.md.
The javascript-conventions folder were created for the translated file.

@nodejs/nodejs-pt：Would you mind helping to have a review？

### CODE REVIEW

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  - stringify
+  - parse
+# title: What is JSON?
+title: O que é JSON?
+difficulty: 5
+layout: knowledge-post.hbs
+---
+
+JavaScript Object Notation ou JSON, é um formato de dados leve que tornou-se
+o padrão de fato para a web. JSON pode ser representado como uma lista de
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
o padrão para a web. JSON pode ser representado como uma lista de
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+valores, ex: um Array, ou um hash de propriedades e valores, ex: um
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
valores, por exemplo, um Array, ou um hash de propriedades e valores, por exemplo, um
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+// um array JSON
+["um", "dois", "tres"]
+
+// um objeto JSON
+{ "um": 1, "dois": 2, "tres": 3 }
+```
+
+## Codificação e Decodificação
+
+O JavaScript fornece 2 métodos, um para codificar estrutura de dados para json
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
O JavaScript fornece 2 métodos. Um para codificar estrutura de dados para JSON
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+e o outro para decodificar json para objetos e arrays javascript.
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
e o outro para decodificar JSON para objetos e arrays javascript.
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+Eles são ambos disponíveis no objeto `JSON` que está disponível no escopo
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
Ambos estão disponíveis no objeto `JSON` que existe no escopo
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  nome: "John Doe",
+  idade: 32,
+  titulo: "Vice Presidente do JavaScript"
+};
+
+const jsonStr = JSON.stringify(dados);
+
+console.log(jsonStr);
+
+// imprimi '{"nome":"John Doe","idade":32,"titulo":"Vice Presidente do JavaScript"}'
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
// imprime '{"nome":"John Doe","idade":32,"titulo":"Vice Presidente do JavaScript"}'
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  "aniversario": new Date('Fri, 26 Jan 2019 07:13:10 GMT'),
+  "getNome": function() {
+      return this.nome;
+  }
+}
+```
+
+Chamar `JSON.parse` com uma string JSON inválida resultará em uma exceção
+SyntaxError. Se você não tem certeza que seu dado JSON é válido,
+você pode anticipar os erros encapsulando a chamada em um block try/catch.
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
você pode antecipar os erros encapsulando a chamada em um bloco try/catch.
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+Observe que os únicos valores complexos permitidos em JSON são objetos e
+arrays. Funções, datas e outros tipos são excluídos. Isso pode não
+parece fazer sentido no começo. Mas lembre-se que o JSON é um formato de dados,
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
parecer fazer sentido no começo. Mas lembre-se que o JSON é um formato de dados,
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+não é um formato para transferir objetos javascript complexos junto com
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
não um formato para transferir objetos javascript complexos junto com
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+## Validadores JSON
+
+Como o JSON se tornou o formato de dados mais amplamente usado, com regras bem definidas, existem muitos validadores disponíveis para auxiliar seu fluxo de trabalho:
+
+- **Validadores online**: Se você está apenas brincando com o JSON ou verificando o JSON de alguém (sem IDEs/editores), os validadores online podem ser de grande ajuda. Por exemplo: [jsonlint.com](https://jsonlint.com) é um bom validador e reformatador online JSON.
+
+* **Pacotes npm**: Se você estiver trabalhando com uma equipe e quiser que um validador JSON seja incluído em seu projeto ou simplesmente goste de automatizar a validação em seu fluxo de trabalho, a grande coleção de pacotes npm estará à sua disposição. Por exemplo: [jsonlint](https://www.npmjs.com/package/jsonlint) é uma versão em JavaScript puro do serviço fornecido em `jsonlint.com`.
+
+- **Plugins para IDEs/editores**: Existem muitos plugins/extensões disponíveis para a maioria dos IDEs/editores que validam o JSON para você. Alguns editores como o `VS Code` vêm com JSON IntelliSense e Validação pronto para uso.
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
- **Plugins para IDEs/editores**: Existem muitos plugins/extensões disponíveis para a maioria dos IDEs/editores que validam o JSON para você. Alguns editores como o `VS Code` vêm com JSON intelliSense e validação pronto para uso.
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-json.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+outras linguagens têm métodos de transferência de hashes e listas nativas
+em objetos JSON stringificados. Aqui está um exemplo rápido em ruby.
+
+```ruby
+require 'json'
+
+dado = { :um => 1 }
+puts dado.to_json
+
+# imprimi "{ \"um\": 1 }"
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-json.md]:
```suggestion
# imprime "{ \"um\": 1 }"
```

**Review by khaosdoctor — CHANGES_REQUESTED** [review]
A few typos

**Comment by 0xthierry** [comment]
Hi, I have fixed right now.

**Review by khaosdoctor — APPROVED** [review]
LGTM 👍

**Comment by khaosdoctor** [comment]
cc: @Maledong @trott

**Comment by MaledongGit** [comment]
@thierrysantoos1  We don't need to do translations for knowledge now, please also see #2278 :)
So I'll close this, and if you have other feedbacks, you can leave them here.

**Pull request merged**